src/Conda.jl

```diff
@@ -104,16 +121,20 @@ end
 "List all installed packages as an array."
 function _installed_packages()
     packages = JSON.parse(readall(`$conda list --json`))
-    regex = r"([\w_-]*)-\d.*"
+    regex = r"^(.*?)-((\d+\.){1,}\d+)[-]*-(.*)$"
+    package_index = Dict{UTF8String, Any}()
     for i in 1:length(packages)
         m = match(regex, packages[i])
         if m != nothing
-            packages[i] = m.captures[1]
+            version = split(m.captures[2], ".")
+            # As julia do not accepts xx.yy.zz.rr version number the last part is removed.
+            # see issue https://github.com/JuliaLang/julia/issues/7282
+            package_index[m.captures[1]] = (VersionNumber(join(version[1:min(length(version), 3)], ".")), packages[i])
         else
             error("Failed parsing string: $(packages[i]). Please open an issue!")
         end
     end
-    return packages
+    return package_index
```

Comment:
If you are going to change the return value, that will mess up other calls to `_installed_packages()` in Conda, no?

Maybe rename this function to `_installed_packages_dict` and define `_installed_packages() = keys(_installed_packages_dict())`?